tests/utils/test_utils.py

```diff
@@ -45,15 +44,20 @@ def __enter__(self):
         return self.path
 
     def __exit__(self, exc_type=None, exc_value=None, traceback=None):
-        if self.is_dir:
-            try:
-                rmtree(self.path)
-            except unittest.SkipTest:
-                # Suppress skip test errors from git.util.rmtree
-                if not exc_type:
-                    raise
-        else:
-            rmfile(self.path)
+        for _ in range(10):
+            if self.is_dir:
+                try:
+                    rmtree(self.path)
+                except unittest.SkipTest:
+                    # Suppress skip test errors from git.util.rmtree
+                    if not exc_type:
+                        raise
+            else:
+                rmfile(self.path)
+
+            if not os.path.exists(self.path):
+                return
+            sleep(0.5)
```

Comment:
why is this required?

Reply:
If the permission error occurs in L50, it will retry requesting removal at most 10 times.